xhr.go

```diff
@@ -58,6 +58,15 @@ const (
 // Request wraps XMLHttpRequest objects. New instances have to be
 // created with NewRequest. Each instance may only be used for a
 // single request.
+//
+// To create a request that behaves in the same way as the top-level
+// Send function with regard to handling binary data, use the
+// following:
+//
+//   req := xhr.NewRequest("POST", "http://example.com")
+//   req.ResponseType = xhr.ArrayBuffer
+//   req.Send([]byte("data"))
+//   b := js.Global.Get("Uint8Array").New(req.Response).Interface().([]byte)
```

Comment:
While this is a logic place for this information to be, I do think it would be more visible and helpful overall to simply move this section to the end of package overview docs.